effectful/ops/syntax.py

```diff
@@ -273,7 +264,7 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> T:
 
 
 @_CustomSingleDispatchCallable
-def defterm(dispatch, value: T) -> Expr[T]:
+def defterm(__dispatch: Callable[[type], Callable[[T], Expr[T]]], value: T):
```

Comment:
Make sure to update signature in `docs/source/effectful.rst`.

Reply:
Why are those handwritten signatures necessary? The code should be the sole source of truth.

Reply:
Sphinx isn't good at detecting callable objects. The workaround seems to be to extend sphinx.

Reply:
See https://github.com/sphinx-doc/sphinx/issues/3783